### PR TITLE
Scaffolding

### PR DESCRIPTION
**Pull request by saraswatpuneet**
- [x]  Basic action
- [x] Make file
- [x] toolchain config file
- [x] fmt config file
- [x] placeholder core library
- [x] Added branch rules and protection rules for main
- [x] License, Contributing, tool-versions

Anything else?

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1,6 @@
+rust 1.64
+make 4.3
+cmake 3.24.1
```
Comment (aramikm) [inline .tool-versions:3]: do we need `cmake` and `protoc` even if when we are not using substrate?

File: core/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+name = "graph-sdk-core"
+edition = "2021"
+homepage = "https://spec.dsnp.org"
+license = "Apache-2.0"
+publish = false
+repository = "https://github.com/libertyDSNP/graph-sdk/"
+version = "0.0.0"
+
+[dependencies]
+wasm-bindgen = {version = "0.2.63", optional = true}
```
Comment (aramikm) [inline core/Cargo.toml:11]: do we want to include wasm dependencies right now or just wait and add whenever we wanted to add support for wasm?

**Review by aramikm — COMMENTED** [review]
Added some questions and I think we might want to add some of the following also:
- LICENSE
- .gitignore

**Review by aramikm — APPROVED** [review]
Looks good!

File: core/Cargo.toml (hunk 2 above)
Comment (saraswatpuneet) [reply core/Cargo.toml:11]: I would say keep it, ideally the rust code should be wasm  compatible and decorated accordingly
can come in later too, so no hard preferences

File: .tool-versions (hunk 1 above)
Comment (saraswatpuneet) [reply .tool-versions:3]: cmake maybe , protoc no, unless any dependency we may have needs it

**Comment by saraswatpuneet** [comment]
@aramikm  Readme, .gitingore and License files are already set in main

**Pull request merged**